Example/GiniVision/ComponentAPICoordinator.swift

```diff
@@ -22,140 +22,178 @@ final class ComponentAPICoordinator: NSObject, Coordinator {
+        }
+    }
+    
     func start() {
         self.setupTabBar()
-        self.newDocumentViewController.delegate = self
+        self.navigationController.delegate = self
         
-        if let document = document {
-            if document.isReviewable {
-                showReviewScreen(withDocument: document)
+        if documentRequests.isEmpty {
+            showCameraScreen()
+        } else {
+            if documentRequests.type == .image {
+                if giniConfiguration.multipageEnabled {
+                    showMultipageReviewScreen()
+                } else {
+                    showReviewScreen()
+                }
             } else {
-                showAnalysisScreen(withDocument: document)
+                showAnalysisScreen()
             }
-        } else {
-            showCameraScreen()
         }
     }
-    
-    // MARK: Show screens
+}
+
+// MARK: Screens presentation
+
+extension ComponentAPICoordinator {
     fileprivate func showCameraScreen() {
-        cameraScreen = self.storyboard.instantiateViewController(withIdentifier: "ComponentAPICamera")
-            as? ComponentAPICameraViewController
+        cameraScreen = CameraViewController(giniConfiguration: giniConfiguration)
         cameraScreen?.delegate = self
-        newDocumentViewController.pushViewController(cameraScreen!, animated: true)
+        cameraScreen?.navigationItem.leftBarButtonItem = UIBarButtonItem(title: "Schließen",
+                                                                         style: .plain,
+                                                                         target: self,
+                                                                         action: #selector(closeComponentAPI))
+        
+        if giniConfiguration.fileImportSupportedTypes != .none {
+            documentPickerCoordinator.delegate = self
+            
+            if giniConfiguration.fileImportSupportedTypes == .pdf_and_images,
+                documentPickerCoordinator.isGalleryPermissionGranted {
+                documentPickerCoordinator.startCaching()
+            }
+            
+            if #available(iOS 11.0, *) {
+                documentPickerCoordinator.setupDragAndDrop(in: cameraScreen!.view)
+            }
+        }
+        navigationController.pushViewController(cameraScreen!, animated: true)
     }
     
-    fileprivate func showReviewScreen(withDocument document: GiniVisionDocument) {
-        reviewScreen = storyboard.instantiateViewController(withIdentifier: "ComponentAPIReview")
-            as? ComponentAPIReviewViewController
+    fileprivate func showMultipageReviewScreen() {
+        navigationController.pushViewController(multipageReviewScreen, animated: true)
+    }
+    
+    fileprivate func showReviewScreen() {
+        guard let document = documentRequests.first?.document else { return }
+        reviewScreen = ReviewViewController(document: document, giniConfiguration: giniConfiguration)
```

Comment:
It is optional since it might not exist when multipage mode is enabled or when a pdf/qrcode is used. 
In a real app it shouldn't be optional (because probably you wouldn't need to keep a reference to it).